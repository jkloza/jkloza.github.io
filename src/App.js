import { useState, useMemo } from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import ResponsiveAppBar from './organisms/app-bar';
import { PageRoutes } from './navigation/routes';
import { ThemeProvider, createTheme } from '@mui/material';
import { getDesignTokens } from './mui/theme';
import { Grid } from '@mui/material';
import { ThemeModeContext } from './context';
import CssBaseline from '@mui/material/CssBaseline';

function App() {
  const [mode, setMode] = useState('dark');

  const colorMode = useMemo(
    () => ({
      // The dark mode switch would invoke this method
      toggleColorMode: () => {
        setMode((prevMode) => (prevMode === 'light' ? 'dark' : 'light'));
      }
    }),
    []
  );

  // Update the theme only if the mode changes
  const theme = useMemo(() => createTheme(getDesignTokens(mode)), [mode]);

  return (
    <ThemeModeContext.Provider value={{ colorMode, mode }}>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        <Router>
          <nav>
            <ResponsiveAppBar />
          </nav>
          <Grid
            container
            justifyContent="center"
            sx={{
              display: 'flex',
              width: '100%',
              alignItems: 'center',
              justifyContent: 'center',
              color: 'text.primary'
            }}>
            <PageRoutes />
          </Grid>
        </Router>
      </ThemeProvider>
    </ThemeModeContext.Provider>
  );
}

export default App;
