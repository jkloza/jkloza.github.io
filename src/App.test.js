import { render, screen } from '@testing-library/react';
import App from './App';

test('renders home page', () => {
  render(<App />);
  const linkElement = screen.getByText(/juli/i);
  expect(linkElement).toBeInTheDocument();
});
