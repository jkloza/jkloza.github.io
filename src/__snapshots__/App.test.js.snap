// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <nav>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-9sroj1-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-1b9hj60-MuiContainer-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-9cyib4-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
            >
              <div
                class="MuiBox-root css-1stx65s"
              >
                <div
                  class="MuiTabs-root css-1ujnqem-MuiTabs-root"
                >
                  <div
                    class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
                    style="overflow: hidden; margin-bottom: 0px;"
                  >
                    <div
                      class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                      role="tablist"
                    >
                      <a
                        aria-selected="true"
                        class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-4jhbo1-MuiButtonBase-root-MuiTab-root"
                        href="/"
                        role="tab"
                        tabindex="0"
                      >
                        Home
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </a>
                      <a
                        aria-selected="false"
                        class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-4jhbo1-MuiButtonBase-root-MuiTab-root"
                        href="/about"
                        role="tab"
                        tabindex="-1"
                      >
                        About
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </a>
                      <a
                        aria-selected="false"
                        class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-4jhbo1-MuiButtonBase-root-MuiTab-root"
                        href="/portfolio"
                        role="tab"
                        tabindex="-1"
                      >
                        Portfolio
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </a>
                    </div>
                    <span
                      class="MuiTabs-indicator css-58tzh6-MuiTabs-indicator"
                      style="left: 0px; width: 0px;"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-130f8nx"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-hc0nzu-MuiSwitch-root"
              >
                <span
                  class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-a4mhny-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-dnkzfa-MuiSwitch-track"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </header>
  </nav>
  <div
    class="MuiGrid-root MuiGrid-container css-fw0cr2-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-1udhcps-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        style="display: flex; justify-content: center; margin-bottom: 10%;"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column css-1qimwsf-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="headshot of juli"
              height="350"
              src="juli-1.png"
              width="350"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <a
                  href="https://github.com/jkloza"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-7klqrd-MuiSvgIcon-root"
                    data-testid="GitHubIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                    />
                  </svg>
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <a
                  href="https://www.linkedin.com/in/juli-kloza-a3180b110/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-7klqrd-MuiSvgIcon-root"
                    data-testid="LinkedInIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                    />
                  </svg>
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <a
                  href="mailto:jkloza205@gmail.com"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-7klqrd-MuiSvgIcon-root"
                    data-testid="MailOutlineIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 css-14a3zbu-MuiGrid-root"
      >
        <div
          style="margin-top: 25%; margin-bottom: 30px;"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-18qle8a-MuiTypography-root"
          >
            greetings! 
            <span
              id="hand-emoji"
            >
              ðŸ‘‹
            </span>
          </h2>
        </div>
        <h5
          class="MuiTypography-root MuiTypography-h5 css-1yzf2g8-MuiTypography-root"
        >
          iâ€™m juli, a 
          <span
            class="MuiTypography-root MuiTypography-h5 css-1r1u2c9-MuiTypography-root"
          >
            full-stack developer
          </span>
           located in boston, massachusetts. welcome to my page! 
        </h5>
        <br />
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-7a9rcu-MuiDivider-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-2 css-12kj135-MuiGrid-root"
      />
    </div>
  </div>
</DocumentFragment>
`;
